lib/src/main/java/graphql/nadel/engine/NadelExecutionContext.kt

```diff
@@ -19,6 +20,8 @@ data class NadelExecutionContext internal constructor(
     val hints: NadelExecutionHints,
     val instrumentationState: InstrumentationState?,
     internal val timer: NadelInstrumentationTimer,
+    internal val deferSupport: NadelIncrementalResultSupport,
+    internal val hydrationDetails: ServiceExecutionHydrationDetails? = null,
```

Comment:
Stores hydration details so that transforms can access it.

In the future this will enable us to handle deferring nested hydrations.